test/SlowTests/Issues/RavenDB_20084.cs

```diff
@@ -47,38 +35,41 @@ public async Task ShouldUpdateBackupInfoIfDatabaseUnloaded()
             customSettings: new Dictionary<string, string>()
             {
                 [RavenConfiguration.GetKey(x => x.Databases.MaxIdleTime)] = "1",
-                [RavenConfiguration.GetKey(x => x.Databases.FrequencyToCheckForIdle)] = "1"
+                [RavenConfiguration.GetKey(x => x.Databases.FrequencyToCheckForIdle)] = "1",
             });
 
         await CreateDatabaseInCluster(databaseName, clusterSize, leaderServer.WebUrl);
 
-        using (var leaderStore = new DocumentStore
-               {
-                   Urls = new[] { leaderServer.WebUrl }, Conventions = new DocumentConventions { DisableTopologyUpdates = true }, Database = databaseName
-               })
+        using (var leaderStore = new DocumentStore())
         {
+            leaderStore.Urls = new[] { leaderServer.WebUrl };
+            leaderStore.Conventions = new DocumentConventions { DisableTopologyUpdates = true };
+            leaderStore.Database = databaseName;
             leaderStore.Initialize();
-            var debugInfo = new List<string>();
-            debugInfo.Add($"Started at: {SystemTime.UtcNow:yyyy-MM-ddTHH:mm:ss.ffffffZ}");
+            var debugInfo = new List<string> { $"Started at: {SystemTime.UtcNow:yyyy-MM-ddTHH:mm:ss.ffffffZ}" };
+
+            leaderServer.ServerStore.DatabasesLandlord.ForTestingPurposesOnly().OnFailedRescheduleNextScheduledActivity = exception =>
+                debugInfo.Add(exception.ToString());
```

Comment:
no concurrency issues here?

Reply:
There shouldn't be any concurrency issues in this test... I'm not reusing the server, but creating a cluster of new servers...
Or am I missing something?

Reply:
My Q to you was if this 'OnFailedRescheduleNextScheduledActivity' can happen at the same time as you are inserting from the test additional debug infos (few lines down).

Reply:
I understand now. Thank you.
I replaced `List<>` with `ConcurrentBag<>`, which should resolve this issue.